crates/lunatic-distributed/src/control/server.rs

```diff
@@ -79,15 +84,59 @@ impl Server {
     }
 }
 
-impl Default for Server {
-    fn default() -> Self {
-        Self::new()
+pub static CTRL_SERVER_NAME: &'static str = "ctrl.lunatic.cloud";
+pub static ROOT_CERT: &'static str = include_str!("../../certs/root.pem");
+static ROOT_KEYS: &'static str = include_str!("../../certs/root.keys.pem");
+
+pub fn root_cert(
+    test_ca: bool,
+    ca_cert: Option<&str>,
+    ca_keys: Option<&str>,
+) -> Result<Certificate> {
+    if test_ca {
+        let key_pair = KeyPair::from_pem(ROOT_KEYS)?;
+        let root_params = CertificateParams::from_ca_cert_pem(ROOT_CERT, key_pair)?;
+        let root_cert = Certificate::from_params(root_params)?;
+        Ok(root_cert)
+    } else {
+        let ca_cert_pem = std::fs::read(Path::new(ca_cert.unwrap()))?;
+        let ca_keys_pem = std::fs::read(Path::new(ca_keys.unwrap()))?;
+        let key_pair = KeyPair::from_pem(std::str::from_utf8(&ca_keys_pem)?)?;
+        let root_params =
+            CertificateParams::from_ca_cert_pem(std::str::from_utf8(&ca_cert_pem)?, key_pair)?;
+        let root_cert = Certificate::from_params(root_params)?;
+        Ok(root_cert)
     }
 }
 
-pub async fn control_server(socket: SocketAddr, cert: String, key: String) -> Result<()> {
-    let mut quic_server = new_quic_server(socket, cert, key)?;
-    let server = Server::new();
+fn sign_node_cert(ca: &Certificate, sign_request: CertificateSigningRequest) -> Result<String> {
+    sign_request
+        .serialize_pem_with_signer(ca)
+        .map_err(|_| anyhow::anyhow!("Error while signing node certificate."))
```

Comment:
Does the error returned by `serialize_pem_with_signer` has some meaningful info?

Reply:
I've added an `Error(String)` response for [RcgenError](https://docs.rs/rcgen/0.9.3/rcgen/enum.RcgenError.html) which implements display.